src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs

```diff
@@ -186,6 +186,12 @@ public async Task<ResultWrapper<PayloadStatusV1>> HandleAsync(ExecutionPayloadV1
                 return NewPayloadV1Result.Invalid(Keccak.Zero);
             }
 
+            if (_poSSwitcher.IsPostMerge(block.Header) && _poSSwitcher.HasInvalidTerminalBlock(out string? invalidTerminalMessage))
```

Comment:
all new payloads should be PostMerge, the parent block of new payload can be terminal block